frontend/cypress/integration/common/workload_logs.ts

```diff
@@ -0,0 +1,122 @@
+    cy.get('#log_hide').type(showText + '{enter}');
+});
+
+When("I choose to show 10 lines of logs", () => {
+    cy.get('#wpl_tailLines-toggle').click();
+    cy.get('#10-0').click();
+});
+
+When("I select only the {string} container", containerName => {
+    let containerListForm = cy.get("[data-test=workload-logs-pod-containers]");
+    containerListForm.get('[type=checkbox]:checked').uncheck();
+
+    let podLabel = containerListForm.get("label").contains(containerName);
+    podLabel.invoke('attr', 'for')
+        .then(labelFor => {
+            cy.get(`#${labelFor}`).check();
+        });
+});
+
+When("I enable visualization of spans", function () {
+    cy.get("#spans-show-").check();
+});
+
+Then("I should see the {string} container listed", (containerName) => {
+    let containerListForm = cy.get("[data-test=workload-logs-pod-containers]");
+
+    let podLabel = containerListForm.get("label").contains(containerName);
+    podLabel.should('have.text', containerName);
+});
+
+Then("the {string} container should be checked", containerName => {
+    let containerListForm = cy.get("[data-test=workload-logs-pod-containers]");
+    let podLabel = containerListForm.get("label").contains(containerName);
+
+    podLabel.invoke('attr', 'for')
+        .then(labelFor => {
+            cy.get(`#${labelFor}`).should('be.checked');
+        });
+});
+
+Then("I should see some {string} pod selected in the pod selector", podNamePrefix => {
+    cy.get('#wpl_pods-toggle').should('include.text', podNamePrefix);
+});
+
+Then("the log pane should only show log lines containing {string}", filterText => {
+    let logPane = cy.get('#logsText');
+    logPane.get('p').each(line => {
+        expect(line).to.contain(filterText);
+    });
+});
+
+Then("the log pane should only show log lines not containing {string}", filterText => {
+    let logPane = cy.get('#logsText');
+    logPane.get('p').each(line => {
+        expect(line).to.not.contain(filterText);
+    });
+});
+
+Then("the log pane should show only {int} lines of logs of each selected container", numberOfLinesPerContainer => {
+    let containerListForm = cy.get("[data-test=workload-logs-pod-containers]");
+    let containersEnabled = containerListForm.get('[type=checkbox]:checked');
+
+    containersEnabled.its('length').then(numContainersEnabled => {
+        let logPane = cy.get('#logsText');
+        logPane.get('p').its('length').should('eq', numContainersEnabled * numberOfLinesPerContainer);
+    });
+});
+
+Then("the log pane should only show logs for the {string} container", containerName => {
+    let containerListForm = cy.get("[data-test=workload-logs-pod-containers]");
+
+    let podLabel = containerListForm.get("label").contains(containerName);
+    podLabel.then($podLabel => {
+        let logColor = $podLabel[0].style.color;
+
+        let logPane = cy.get('#logsText');
+        logPane.get('p').each(line => {
+            expect(line[0].style.color).to.equal(logColor);
+        })
```

Comment:
Is there a way to check all the text at once as opposed to line by line? Might not be. I think the default log limit is 100 so probably not a big deal.

Reply:
Hmm, there is an `all` chainer for  the `expect` assertion. Let me try it...

Reply:
Hmm... Nope, that `all` chainer only lets you follow up with `keys` or `member`. Not any property (here I would want `all.css`).